compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1204,6 +1254,122 @@ fn normalize_to_error<'a, 'tcx>(
     Normalized { value: new_value, obligations: vec![trait_obligation] }
 }
 
+/// Confirm and normalize the given inherent projection.
+#[instrument(level = "debug", skip(selcx, param_env, cause, obligations))]
+pub fn normalize_inherent_projection<'a, 'b, 'tcx>(
+    selcx: &'a mut SelectionContext<'b, 'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    alias_ty: ty::AliasTy<'tcx>,
+    cause: ObligationCause<'tcx>,
+    depth: usize,
+    obligations: &mut Vec<PredicateObligation<'tcx>>,
+) -> Ty<'tcx> {
+    let tcx = selcx.tcx();
+
+    if !tcx.recursion_limit().value_within_limit(depth) {
+        // FIXME(inherent_associated_types): Is there a better way to deal with overflow errors here?
```

Comment:
Would you be fine with me passing the `AliasTy` directly to `report_overflow_error` (the diagnostic's gonna say ``overflow evaluating the requirement `T::P` ``)? Otherwise I'd need to smh. make a fake infer ty via `tcx.mk_ty_var` for the RHS of the `ProjectionPredicate`. This all feels like a lie :sweat_smile: